examples/Example1/example1.cu

```diff
@@ -0,0 +1,188 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include <curand.h>
+#include <curand_kernel.h>
+#include <iostream>
+#include <iomanip>
+#include <stdio.h>
+
+#include <AdePT/BlockData.h>
+
+#include "process.h"
+#include "process_list.h"
+#include "pair_production.h"
+#include "energy_loss.h"
+
+#include "track.h"
+
+#include <AdePT/MParray.h>
+
+// some simple scoring
+struct Scoring {
+  adept::Atomic_t<int> secondaries;
+  adept::Atomic_t<float> totalEnergyLoss;
+
+  VECCORE_ATT_HOST_DEVICE
+  Scoring() {}
+
+  VECCORE_ATT_HOST_DEVICE
+  static Scoring *MakeInstanceAt(void *addr)
+  {
+    Scoring *obj = new (addr) Scoring();
+    return obj;
+  }
+};
+
+// kernel select processes based on interaction lenght and put particles in the appropriate queues
+__global__ void DefinePhysicalStepLength(adept::BlockData<track> *block, process_list** proclist, adept::MParray **queues, curandState_t *states)
+{
+  int n = block->GetNused() + block->GetNholes();
+
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n; i += blockDim.x * gridDim.x) {
+    
+    // skip particles that are already dead
+    if ((*block)[i].status == dead) continue;
+
+    (*proclist)->GetPhysicsInteractionLength(i, block, states); // return value (if step limited by physics or geometry) not used for the moment
+    // now, I know which process wins, so I add the particle to the appropriate queue
+    queues[(*block)[i].current_process]->push_back(i);
```

Comment:
right now, I think we always have a process in this simple example. Just curious, how would this extend to transportation?

Reply:
transportation is just yet another process (it gives the distance to the next boundary as 'interaction lenght'. and it's a 'along step process' that always takes place (you can have several along step process). The distance to the boundary is used to check if the step is limited by physics or geometry .

Reply:
This is of course how it is done in Geant4, in our case we are not bound to implement as a process having the same interface, but reduction to get the minimum proposed step (geometry vs. a given physics process) has to be done at some point for every step.